demos/mqtt/mqtt_demo_connection_sharing/mqtt_demo_connection_sharing.c

```diff
@@ -0,0 +1,1601 @@
+        }
+    }
+
+    return returnStatus;
+}
+
+/*-----------------------------------------------------------*/
+
+static int disconnectMqttSession( MQTTContext_t * pMqttContext )
+{
+    MQTTStatus_t mqttStatus = MQTTSuccess;
+    int returnStatus = EXIT_SUCCESS;
+
+    assert( pMqttContext != NULL );
+
+    /* Send DISCONNECT. */
+    mqttStatus = MQTT_Disconnect( pMqttContext );
+
+    if( mqttStatus != MQTTSuccess )
+    {
+        LogError( ( "Sending MQTT DISCONNECT failed with status = %u.",
+                    mqttStatus ) );
+        returnStatus = EXIT_FAILURE;
+    }
+
+    return returnStatus;
+}
+
+/*-----------------------------------------------------------*/
+
+static int subscribeToTopic( MQTTContext_t * pMqttContext,
+                             const char * pTopicFilter,
+                             uint16_t topicFilterLength )
+{
+    int returnStatus = EXIT_SUCCESS;
+    MQTTStatus_t mqttStatus;
+    MQTTSubscribeInfo_t pSubscriptionList[ 1 ];
+
+    assert( pMqttContext != NULL );
+
+    /* Start with everything at 0. */
+    ( void ) memset( ( void * ) pSubscriptionList, 0x00, sizeof( pSubscriptionList ) );
+
+    /* This example subscribes to only one topic and uses QOS1. */
+    pSubscriptionList[ 0 ].qos = MQTTQoS1;
+    pSubscriptionList[ 0 ].pTopicFilter = pTopicFilter;
+    pSubscriptionList[ 0 ].topicFilterLength = topicFilterLength;
+
+    /* Generate packet identifier for the SUBSCRIBE packet. */
+    globalSubscribePacketIdentifier = MQTT_GetPacketId( pMqttContext );
+
+    /* Send SUBSCRIBE packet. */
+    mqttStatus = MQTT_Subscribe( pMqttContext,
+                                 pSubscriptionList,
+                                 sizeof( pSubscriptionList ) / sizeof( MQTTSubscribeInfo_t ),
+                                 globalSubscribePacketIdentifier );
+
+    if( mqttStatus != MQTTSuccess )
+    {
+        LogError( ( "Failed to send SUBSCRIBE packet to broker with error = %u.",
+                    mqttStatus ) );
+        returnStatus = EXIT_FAILURE;
+    }
+    else
+    {
+        LogInfo( ( "SUBSCRIBE sent for topic %.*s to broker.\n\n",
+                   topicFilterLength,
+                   pTopicFilter ) );
+    }
+
+    return returnStatus;
+}
+
+/*-----------------------------------------------------------*/
+
+static int unsubscribeFromTopic( MQTTContext_t * pMqttContext,
+                                 const char * pTopicFilter,
+                                 uint16_t topicFilterLength )
+{
+    int returnStatus = EXIT_SUCCESS;
+    MQTTStatus_t mqttStatus;
+    MQTTSubscribeInfo_t pSubscriptionList[ 1 ];
+
+    assert( pMqttContext != NULL );
+
+    /* Start with everything at 0. */
+    ( void ) memset( ( void * ) pSubscriptionList, 0x00, sizeof( pSubscriptionList ) );
+
+    /* This example subscribes to and unsubscribes from only one topic
+     * and uses QOS2. */
+    pSubscriptionList[ 0 ].qos = MQTTQoS2;
```

Comment:
There's no point in subscribing with QoS2 if the all the publishes are only QoS1

Reply:
Removed this statement altogether as it is irrelevant to the **unsubscribe** operation